text/0111-feature-flags.md

```diff
@@ -154,9 +154,30 @@ describe('foo', () => {
 });
 ```
 
+Note that this will only work for features behind flags that are not used during the initialization of the engine.
+
+```js
+// Works when run as part of initialization
+const foo = () => {
+    if (ENABLE_FOO) {
+        return 1;
+    } else {
+        return 2;
+    }
+};
+
+// Doesn't work when run as part of initialization
+let foo;
+if (ENABLE_FOO) {
+    foo = () => 1;
+} else {
+    foo = () => 2;
+}
+```
```

Comment:
the examples above don't demonstrate their relationship to the engine initialization

Reply:
The former evaluates `ENABLE_FOO` when the function `foo` is invoked so it'll follow the right path if the flag is toggled during a test. The latter evaluates `ENABLE_FOO` during initialization so by the time we toggle the flag in a test, it's too late in that it won't make a difference.